include/deal.II/numerics/data_out_dof_data.h

```diff
@@ -591,19 +591,18 @@ namespace internal
  *
  * @ingroup output
  */
-template <typename DoFHandlerType,
+template <int dim,
           int patch_dim,
-          int patch_space_dim = patch_dim>
-class DataOut_DoFData : public DataOutInterface<patch_dim, patch_space_dim>
+          int spacedim       = dim,
+          int patch_spacedim = patch_dim>
+class DataOut_DoFData : public DataOutInterface<patch_dim, patch_spacedim>
```

Comment:
We talked about the order of template arguments -- I think this is fine as is, the class is internal anyway, so let's just leave it as is.